KIPs/kip-71.md

```diff
@@ -0,0 +1,386 @@
+	type: int = 0x32
+	nonce: int = 0
+	gas_price: int = 0
+	gas: int = 0
+	to: int = 0
+	value: int = 0
+	from: int = 0
+	input: bytes = bytes()
+	fee_ratio: int = 1
+	tx_signatures: List[int] = field(default_factory=list)
+	fee_payer: int = 0 
+	fee_payer_signatures: List[int] = field(default_factory=list)
+
+
+# In Klaytn, Ethereum transaction types are supported by adding EthereumTxTypeEnvelope(0x78).
+# RawTransaction : EthereumTxTypeEnvelope || EthereumTransactionType || TransactionPayload
+# (|| is the byte/byte-array concatenation operator.)
+# In this proposal, those concatenation is intentionally ommited for simplification.
+
+# Transaction2930 in Ethereum. 
+@dataclass
+class TxTypeEthereumAccessList:
+	type: int = 0x7801
+	chain_id: int = 0
+	nonce: int = 0
+	gas_price: int = 0
+	gas: int = 0
+	to: int = 0
+	value: int = 0
+	data: bytes = bytes()
+	access_list: List[Tuple[int, List[int]]] = field(default_factory=list)
+	v: int = 0
+	r: int = 0
+	s: int = 0
+
+# Transaction1559 in Ethereum.
+@dataclass
+class TxTypeEthereumDynamicFee:
+	type: int = 0x7802
+	chain_id: int = 0
+	nonce: int = 0
+	gas_tip_cap: int = 0
+	gas_fee_cap: int = 0
+	gas: int = 0
+	to: int = 0
+	value: int = 0
+	data: bytes = bytes()
+	access_list: List[Tuple[int, List[int]]] = field(default_factory=list)
+	v: int = 0
+	r: int = 0
+	s: int = 0
+
+
+EthereumTransactions = Union[TxTypeEthereumDynamicFee, TxTypeEthereumAccessList]
+
+Transaction = Union[TxTypeLegacyTransaction, TxTypeFeeDelegatedSmartContractExecution, TxTypeFeeDelegatedSmartContractExecutionWithRatio, EthereumTransactions]
+
+# TODO: transaction accounting part
+@dataclass
+class NormalizedTransaction:
+	signer_address: int = 0
+	signer_nonce: int = 0
+	max_fee_per_gas: int = 0
+	gas_limit: int = 0
+	to: int = 0
+	value: int = 0
+	data: bytes = bytes()
+	fee_payer_address: int = 0
+	fee_ratio: int = 1
+
+@dataclass
+class Block:
+	hash: int = 0
+	parent_hash: int = 0
+	base_fee_per_gas: int = 0
+	block_score: int = 0
+	extra_data: bytes = bytes()
+	gas_used: int = 0 
+	governance_data: bytes = bytes()
+	logs_bloom: int = 0
+	number: int = 0
+	transaction_receipt_root: int = 0
+	reward: int = 0
+	state_root: int = 0
+	timestamp: int = 0
+	timestamp_FoS: int = 0
+	transaction_root: int = 0
+	nonce: int = 0
+	committee: List[int] = field(default_factory=list)
+	proposer: int = 0 
+	size: int = 0 
+
+
+@dataclass
+class Account:
+	type: int = 0
+	nonce: int = 0
+	humanReadable: bool = False 
+	address: int = 0
+	key: int = 0 
+	balance: int = 0
+	storage_root: int = 0
+	code_hash: int = 0
+	code_format: int = 0
+	vm_version: int = 0
+
+INITIAL_FORK_BLOCK_NUMBER = 107806544 # TBD
+BASE_FEE_DELTA_REDUCING_DENOMINATOR = 36 # Max basefee change: 5% per block 
+LOWER_BOUND_BASE_FEE = 25000000000 # 25ston
+UPPER_BOUND_BASE_FEE = 750000000000 # 750ston
+GAS_TARGET = 30000000 
+BLOCK_GAS_LIMIT = 84000000 # implicit block gas limit to enforce the max basefee change rate
```

Comment:
It seems that others can mislead Klaytn have block gas limit in the future. I think we can choose any other name for this. `MAX_BLOCK_GAS_USED_FOR_BASE_FEE`?

Reply:
I also think the name, `BLOCK_GAS_LIMIT` can be confusing in ethereum's one. We use it only for calculating `BASE_FEE`